src/main/telemetry/smartport.c

```diff
@@ -365,9 +365,13 @@ static void processMspPacket(mspPacket_t* packet)
 {
     initSmartPortMspReply(0);
 
-    if (mspFcProcessCommand(packet, &smartPortMspReply, NULL) == MSP_RESULT_ERROR) {
+    mspPostProcessFnPtr mspFcProcessFn = NULL;
+    if (mspFcProcessCommand(packet, &smartPortMspReply, &mspFcProcessFn) == MSP_RESULT_ERROR) {
         sbufWriteU8(&smartPortMspReply.buf, SMARTPORT_MSP_ERROR);
     }
+    if (mspFcProcessFn) {
+        mspFcProcessFn(NULL);
```

Comment:
Because you are calling this with a `NULL` parameter here, you need to add a check to all the functions that are used as a `mspPostProcessFnPtr` to make sure they don't execute if the serial port parameter is actually used. Otherwise the flight controller will end up in an undefined state if one of the commands using these functions is received.